### PR TITLE
[14.0] Add  shopfloor_reception_refund_return

### PR DESCRIPTION
**Pull request by mmequignon**
This module, atm, just sets `to_refund` on return moves created by shopfloor.

### CODE REVIEW

File: shopfloor_account/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Glue module between `shopfloor` and `stock_account`
```
Comment (TDu) [inline shopfloor_account/readme/DESCRIPTION.rst]: Maybe explain why ?

**Review by TDu — APPROVED** [review]

File: shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+except (ImportError, IOError) as err:
+    _logger.debug(err)
+
+
+def move_fields_to_new_module(cr):
+    # is_shopfloor_created has been wrongly put in shopfloor_reception.
+    # this script moves it in shopfloor.
+    openupgrade.update_module_moved_fields(
+        cr, "stock.picking", "is_shopfloor_created", "shopfloor_reception", "shopfloor"
+    )
```
Comment (sebalix) [inline shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py]: <s>Does it work well if `shopfloor` is upgraded first (which will create the `stock_picking.is_shopfloor_created` column in DB), then comes the upgrade of `shopfloor_reception` with this script?</s>
EDIT: it's OK, I said nothing

File: shopfloor_account/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "name": "Shopfloor Account",
+    "summary": "Glue module between shopfloor and stock_account",
+    "version": "14.0.1.0.0",
+    "category": "Inventory, Accounting",
+    "website": "https://github.com/OCA/wms",
+    "author": "Camptocamp SA, Odoo Community Association (OCA)",
+    "maintainers": ["mmequignon"],
+    "license": "AGPL-3",
+    "installable": True,
+    "auto_install": False,
```
Comment (sebalix) [inline shopfloor_account/__manifest__.py]:
```suggestion
    "auto_install": True,
```

**Review by sebalix — COMMENTED** [review]
`auto_install` flag to fix

File: shopfloor_account/__manifest__.py (hunk 3 above)
Comment (jbaudoux) [reply shopfloor_account/__manifest__.py]: I would keep auto install False as this is not a glue module but an optional module

**Review by jbaudoux — CHANGES_REQUESTED** [review]
This is not a glue module.
This is an optional module that enables the refund of all the created reception returns

Can you rename the module to `shopfloor_reception_refund_returns` ?

**Comment by simahawk** [comment]
Final "s" should be removed from the name :wink:

**Review by jbaudoux — APPROVED** [review]
@TDu @sebalix I changed the module name, rewrote commit messages. Ok for you?

**Comment by jbaudoux** [comment]
> Final "s" should be removed from the name wink

removed

File: shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from openupgradelib import openupgrade
+except (ImportError, IOError) as err:
+    _logger.debug(err)
```
Comment (jbaudoux) [inline shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py]: Why `try?. If it cannot be imported, the migration will fail. Why not adding it in the repo requirements.txt ?

**Review by jbaudoux — CHANGES_REQUESTED** [review]

File: shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+
+def move_fields_to_new_module(cr):
+    # is_shopfloor_created has been wrongly put in shopfloor_reception.
+    # this script moves it in shopfloor.
+    openupgrade.update_module_moved_fields(
+        cr, "stock.picking", "is_shopfloor_created", "shopfloor_reception", "shopfloor"
```
Comment (jbaudoux) [inline shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py]: It raises an error:
```
shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py", line 17, in move_fields_to_new_module
    openupgrade.update_module_moved_fields(
  File "python3.8/site-packages/openupgradelib/openupgrade.py", line 2676, in update_module_moved_fields
    do_raise("moved_fields %s must be a tuple or list!" % moved_fields)
```

File: shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py (hunk 4 above)
Comment (TDu) [reply shopfloor_reception/migrations/14.0.2.1.0/pre-migration.py]: Agree. Removed the try and added the dependency.

**Comment by TDu** [comment]
Rebased and added the dependency on openupgradelib.
But also IMO the shopfloor module should have its version bumped (a new field is added).
So shouldn't the script be actually in the shopfloor module ?

File: shopfloor_reception/__manifest__.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "Shopfloor Reception",
     "summary": "Reception scenario for shopfloor",
-    "version": "14.0.2.3.0",
+    "version": "14.0.2.4.0",
```
Comment (jbaudoux) [inline shopfloor_reception/__manifest__.py:4]: the migration script must have the same version

Comment (TDu) [reply shopfloor_reception/__manifest__.py:4]: changed

**Review by jbaudoux — APPROVED** [review]

**Comment by simahawk** [comment]
@mmequignon pls rebase

**Comment by TDu** [comment]
@simahawk  Can we reopen this one

**Comment by simahawk** [comment]
I cannot. It says "the branch was force-pushed or recreated". You'll have to open a new one.

**Comment by TDu** [comment]
I have opened https://github.com/OCA/wms/pull/669 to follow up

**Pull request closed without merging**